jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -63,99 +63,93 @@ async def job_script_create(
     status_code=status.HTTP_201_CREATED,
     response_model=JobScriptResponse,
     description="Endpoint for job script creation",
+    dependencies=[Depends(file_services(file_service, template_file_service))],
 )
 async def job_script_create_from_template(
     create_request: JobScriptCreateRequest,
     render_request: RenderFromTemplateRequest,
     id_or_identifier: int | str = Path(...),
-    session: AsyncSession = Depends(db_session),
-    job_script_service: TableService = Depends(job_script_service),
-    job_script_file_service: FileService = Depends(job_script_files_service),
-    template_service: TableService = Depends(template_service),
-    template_file_service: FileService = Depends(template_files_service),
-    token_payload: TokenPayload = Depends(guard.lockdown(Permissions.JOB_SCRIPTS_EDIT)),
+    secure_session: SecureSession = Depends(
+        secure_services(
+            Permissions.JOB_SCRIPTS_EDIT,
+            services=[crud_service, template_crud_service, file_service, template_file_service],
+        )
+    ),
 ):
     """Create a new job script from a job script template."""
     logger.info(f"Creating a new job script from {id_or_identifier=} with {create_request=}")
 
-    identity_claims = IdentityClaims.from_token_payload(token_payload)
-    if identity_claims.email is None:
+    if secure_session.identity_payload.email is None:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST,
             detail="The token payload does not contain an email",
         )
 
-    if not render_request.template_output_name_mapping:
+    base_template = cast(JobScriptTemplate, await template_crud_service.get(id_or_identifier))
+
+    required_map = render_request.template_output_name_mapping
+    entrypoint_key = list(required_map.values())[0]
```

Comment:
I'm wondering if I did a good call here by using the first element on the map as the entry point. I intended to avoid duplication by having a map for the entrypoint (with just one element) and another for the supporting files. On the other hand, I'm not entirely sure that we can trust that the order will be kept by all kinds of tools making requests to Jobbergate.

Removing the prefix .j2 or .jinja2 from the templates should be enough since they are already propper named. The filename mapping is needed for backward compatibility.

Reply:
I think this is something we should possibly revist. I agree that it's not robust, but we need to consider this more and possibly change it later.